app/services/ChargeHistoryService.scala

```diff
@@ -68,4 +69,32 @@ class ChargeHistoryService @Inject()(chargeHistoryConnector: ChargeHistoryConnec
     }
   }
 
+  def getReviewAndReconcileCredit(chargeItem: ChargeItem,
+                                  chargeDetailsforTaxYear: FinancialDetailsModel,
+                                  reviewAndReconcileEnabled: Boolean): Option[ReviewAndReconcileCredit] = {
+
+    if (reviewAndReconcileEnabled)
+      chargeDetailsforTaxYear
+        .getPairedDocumentDetails()
```

Comment:
* if you can add comment/tag(?) around method getPairedDocumentDetails in FinancialDetailsModel as deprecated, as we are planning to hide DocumentDetail within FinancialDetailsModel and rely fully on ChargeItem in our domain instead: here is a story for reference => https://jira.tools.tax.service.gov.uk/browse/MISUV-3816;
Please add similar method to FinancialDetailsModel which will return ChargeItem and rely on this method in this part of the logic;

Reply:
also I am not entirely understand how chargeItem is related to charges in chargeDetailsforTaxYear; as I see they are joined purely on the fact that we are filtering out POA_credits from chargeDetailsforTaxYear